ts/components/session/SessionSeedModal.tsx

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+
+import { SessionModal } from './SessionModal';
+import { SessionButton } from './SessionButton';
+
+interface Props {
+  onClose: any;
+}
+
+interface State {
+  error: string;
+  loadingPassword: boolean;
+  loadingSeed: boolean;
+  seed: string;
+  hasPassword: boolean | null;
+  passwordHash: string;
+  passwordValid: boolean;
+}
+
+export class SessionSeedModal extends React.Component<Props, State> {
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      error: '',
+      loadingPassword: true,
+      loadingSeed: true,
+      seed: '',
+      hasPassword: null,
+      passwordHash: '',
+      passwordValid: false,
+    };
+
+    this.copySeed = this.copySeed.bind(this);
+    this.getSeed = this.getSeed.bind(this);
+    this.confirmPassword = this.confirmPassword.bind(this);
+    this.checkHasPassword = this.checkHasPassword.bind(this);
+    this.onEnter = this.onEnter.bind(this);
+
+    window.addEventListener('keyup', this.onEnter);
+  }
+
+  public render() {
+    const i18n = window.i18n;
+    const { onClose } = this.props;
+
+    const maxPasswordLen = 64;
+
+    this.checkHasPassword();
+    void this.getSeed();
```

Comment:
You should remove the void and add ignore() at the end. that's the way I've been told to ignore a promise